### PR TITLE
BAAS-18471: Add float integers support in the isSafeInteger native function

### PR DESCRIPTION
**Pull request by jwongmongodb**
[BAAS-18471](https://jira.mongodb.org/browse/BAAS-18471)

The main issue is that the dependency is-odd is setting a value calling math_abs() which returns a float value. The below is the detailed explanation:
1. The is-odd package calls math.abs() which converts string and negative numbers to the positive number.
2. The math.abs() has a native function implementation in goja which always returns valueFloat.
3. Then the isSafeInteger() is called from the is-odd package which leads to the goja problem with the valueFloat

Summary of the change:
Made a change to isSafeInteger to handle a valueFloat since isInteger() already has a case to handle floats ([code link](https://github.com/mongodb-forks/goja/blob/realm/builtin_number.go#L188)). We can add a check if the value coming in isSafeInteger is a valueFloat, to call isInteger() to see if the floatValue is an integer and then check if its within the max int range.

After making this change, the is-odd npm package will correctly work with the math.abs value being passed into isSafeInteger() (as well as all future usage patterns)

### CODE REVIEW

File: builtin_number.go [hunk 1]
```diff
@@ -211,6 +211,10 @@ func (r *Runtime) number_isSafeInteger(call FunctionCall) Value {
 	if i, ok := call.Argument(0).(valueInt64); ok && i >= -(maxInt-1) && i <= maxInt-1 {
```
Comment (Gabri3l) [inline builtin_number.go:211]:
```suggestion
	if i, ok := arg.(valueInt64); ok && i >= -(maxInt-1) && i <= maxInt-1 {
```

File: builtin_number.go [hunk 2]
```diff
@@ -211,6 +211,10 @@ func (r *Runtime) number_isSafeInteger(call FunctionCall) Value {
...
 		return valueTrue
 	}
+	// valueFloat can come in from other nativefuncs like math_abs, this will make sure that the valueFloat is a valid integer
+	if i, ok := call.Argument(0).(valueFloat); ok && i >= -(maxInt-1) && i <= maxInt-1 {
```
Comment (Gabri3l) [inline builtin_number.go:215]:
```suggestion
	if i, ok := arg.(valueFloat); ok && i >= -(maxInt-1) && i <= maxInt-1 {
```

File: builtin_number_test.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package goja
+
+import "testing"
+
+func TestIsSafeInteger(t *testing.T) {
+	const SCRIPT = `
+	var maxInt = 9007199254740991
+    var overflowInt = 9007199254740992
+    var maxIntFloat = 9007199254740991.0
```
Comment (Gabri3l) [inline builtin_number_test.go:9]:
```suggestion
        var overflowInt = 9007199254740992
        var maxIntFloat = 9007199254740991.0
```

**Review by Gabri3l — COMMENTED** [review]
Just some super tiny things, welcome to goja!

File: builtin_number.go (hunk 1 above)
Comment (Gabri3l) [reply builtin_number.go:211]: I know this was existing code, might as well change it to stay consistent.

**Comment by Gabri3l** [comment]
Oh boy this is trippy, all your tests are great but in JS if you do:

```js
Number.isSafeInteger(Math.abs(2.2)) // this is False
Number.isSafeInteger(Math.abs(2)) // this is True
```

Gotta love JS! I think the underlying Math.abs doesn't work as per-spec but I think we should be ok

**Comment by jwongmongodb** [comment]
> Oh boy this is trippy, all your tests are great but in JS if you do:
> 
> ```js
> Number.isSafeInteger(Math.abs(2.2)) // this is False
> Number.isSafeInteger(Math.abs(2)) // this is True
> Number.isSafeInteger(2.2) // this is False
> ```
> 
> Gotta love JS! I think the underlying Math.abs doesn't work as per-spec so based on my tests up here it almost seems like `isSafeInteger` is working as expected and we might need to change Math.abs.

Yea, I didn't want to make a change to Math.abs() native override since that is a bigger risk. I made a comment on ticket regarding the decision of making this type of change from the isSafeInteger func.

**Pull request merged**